### PR TITLE
Rename *TilesetType to *TileAnimations

### PR DESCRIPTION
**Pull request by SatoMew**
The current label and comments seem to be based on which values are set in individual maps, but they don't actually reflect what is controlled by this property or what real effect does each value have.

### CODE REVIEW

**Review by Rangi42 — APPROVED** [review]
Looks good to me. (It and a few recent commits will need merging into pokeyellow too.)

**Comment by dannye** [comment]
I think `wSavedTilesetType` and `sTilesetType` need to be similarly renamed.

**Comment by dannye** [comment]
I think [these constants](https://github.com/pret/pokered/blob/master/constants/map_data_constants.asm#L12) are less self-explanatory now.

**Comment by SatoMew** [comment]
I'll update those as well, thanks!

**Comment by SatoMew** [comment]
This PR will also rename the unreferenced fanfare accordingly.

File: constants/map_data_constants.asm [hunk 1]
```diff
@@ -9,8 +9,8 @@ MAP_BORDER EQU 3
 	shift_const SOUTH ; 4
 	shift_const NORTH ; 8
 
-; tileset environments
+; flower and water tile animations
```
Comment (Rangi42) [inline constants/map_data_constants.asm:12]: How about: `TILEANIM_NONE`, `TILEANIM_WATER`, `TILEANIM_WATER_FLOWER`

File: hram.asm [hunk 2]
```diff
@@ -277,11 +277,11 @@ hFrameCounter:: db ; decremented every V-blank (used for delays)
 ; you can detect that the V-blank handler has run since then.
 hVBlankOccurred:: db
 
-; 00 = indoor
-; 01 = cave
-; 02 = outdoor
-; this is often set to 00 in order to turn off water and flower BG tile animations
-hTilesetType:: db
+; Controls whether flower and water tiles are animated.
+; 00 = flowers and water are static (breaks Surf)
```
Comment (Rangi42) [inline hram.asm:281]:
```
; Controls which tiles are animated.
; 0 = no animations (breaks Surf)
; 1 = water tile $14 is animated
; 2 = water tile $14 and flower tile $03 are animated
```

File: constants/map_data_constants.asm (hunk 1 above)
Comment (SatoMew) [reply constants/map_data_constants.asm:12]: Sounds good to me.

File: hram.asm (hunk 2 above)
Comment (SatoMew) [reply hram.asm:281]: Specifying the tile IDs looks reasonable so I will adjust the comment as suggested.

**Review by Rangi42 — APPROVED** [review]

**Review by dannye — APPROVED** [review]

File: audio/sfx/unreferenced_fanfare.asm [hunk 3]
```diff
@@ -1,4 +1,5 @@
-SFX_Unused2_Ch5:
+; This corresponds to SFX_KEY_ITEM in pokegold and pokecrystal.
+SFX_UnreferencedFanfare_Ch5:
```
Comment (Rangi42) [inline audio/sfx/unreferenced_fanfare.asm:2]: Please make this "UnusedFanfare"; the current Gen 2 convention is to keep `; unreferenced` in comments. (pokered has a couple of those but needs more.)

Comment (SatoMew) [reply audio/sfx/unreferenced_fanfare.asm:2]: Will do, thanks!

**Review by dannye — CHANGES_REQUESTED** [review]
Sorry for going in circles, but can you change the filename `unreferenced_fanfare.asm` to `unused_fanfare.asm` to match the labels?

**Comment by SatoMew** [comment]
Sure.

**Pull request merged**